src/ui/component/dropDown/dropDown.tsx

```diff
@@ -1,168 +1,152 @@
-import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import type { FC } from 'react'
-import { useTranslation } from 'react-i18next'
-import classnames from 'classnames'
-import { useOnClickOutside } from '@/ui/hook/useOnClickOutside'
-import { Icon } from '@/ui/component/icon/icon'
+import { useMemo, useCallback, useState } from 'react'
+import { Collapsible } from '@/ui/component/collapsible/collapsible'
+import { SearchBar } from '@/ui/component/searchbar/searchbar'
+import { isSubstringOf } from '@/util/util'
+import { DynamicCheckbox } from '@/ui/view/dataverse/component/dynamicCheckbox/dynamicCheckbox'
+import { NoResultFound } from '@/ui/view/dataverse/component/noResultFound/noResultFound'
+import { Tag } from '../tag/tag'
 import './dropDown.scss'
 
-export type SubOption = Omit<Option, 'subOptions'> & { parentOptionValue: string }
-
-export type Option = {
-  label: string
+export type SelectionOption = {
+  selected: boolean
   value: string
-  subOptions?: SubOption[]
+  name: string
+  disabled: boolean
 }
 
-type DropDownMenu = {
-  label: string
-  menuOptions: Option[]
-}
+type SelectionItemType = 'checkbox'
 
 type DropDownProps = {
-  options: Option[]
-  onChange: (selectedOption: string, selectedSubOption?: string) => void
-  value: string
+  filterName: string
+  filterValues: string[]
+  searchPlaceholder: string
+  selectionType: SelectionItemType
+  maxSearchResults?: number
 }
 
 // eslint-disable-next-line max-lines-per-function
-export const DropDown: FC<DropDownProps> = ({ options, onChange, value }) => {
-  const { t } = useTranslation('common')
-  const [menuOpened, setMenuOpened] = useState<boolean>(false)
-  const [focusedMenuIndex, setFocusedMenuIndex] = useState<number>(0)
-  const [dropDownMenus, setDropDownMenus] = useState<DropDownMenu[]>([])
-  const menuContainerRef = useRef<HTMLDivElement | null>(null)
-  const dropDownMenusRefs = useRef<HTMLDivElement[]>([])
-
-  const toggleMenu = useCallback(() => {
-    setMenuOpened(!menuOpened)
-  }, [menuOpened])
-
-  const clickOutsideHandler = useCallback(() => {
-    setMenuOpened(false)
-  }, [])
-
-  useOnClickOutside(menuContainerRef, clickOutsideHandler)
-
-  const selectOption = useCallback(
-    (optionValue: string, subOptionValue?: string) => {
-      onChange(optionValue, subOptionValue)
-      toggleMenu()
-    },
-    [onChange, toggleMenu]
+export const DropDown: FC<DropDownProps> = ({
+  filterName,
+  filterValues,
```

Comment:
Here you want to control the component by giving back the selected options, so maybe you could handle directly a list of them, that looks much more explicit, IMHO.